chart,salt,build: Migrate to fluent-bit fluent chart

TeddyAndrieux opened the pull request:
Migrate from grafana fluent-bit deprecated helm chart to the fluent-bit
helm chart from fluent version 0.19.19

Move from `fluent-bit-plugin-loki` to `fluent-bit` version 1.8.12.

Rewrite fluent-bit output from config as it's not exactly the same
between `grafana-loki` and `loki`.

We also enable HTTP_Server back as it seems to work properly with this
version.

Fixes: #3435

TeddyAndrieux committed:
salt: Increase fluent-bit buffer limit from 5MB to 50MB

Also bump Kubernetes ressources requests for memory

We also bump the number of workers used by fluent-bit.

Re-render using:
```
./charts/render.py fluent-bit --namespace metalk8s-logging \
  charts/fluent-bit.yaml charts/fluent-bit/ \
  > salt/metalk8s/addons/logging/fluent-bit/deployed/chart.sls
```

TeddyAndrieux committed:
chart,salt,build: Migrate to fluent-bit fluent chart

Migrate from grafana fluent-bit deprecated helm chart to the fluent-bit
helm chart from fluent:

```
rm -rf charts/fluent-bit
helm repo add fluent https://fluent.github.io/helm-charts
helm repo update
helm fetch -d charts --untar fluent/fluent-bit
```

Change image from `fluent-bit-loki-plugin` from grafana repository to
1.8.12 `fluent-bit` from fluent repository as Loki output plugin is
built-in.

Migrate fluent-bit option file to work with the new fluent-bit helm
template.

Rewrite fluent-bit output from config as it's not exactly the same
between `grafana-loki` and `loki`.

We also enable HTTP_Server back as it seems to work properly with this
version.

Render chart to salt state using:

```
./charts/render.py fluent-bit --namespace metalk8s-logging \
  charts/fluent-bit.yaml charts/fluent-bit/ \
  > salt/metalk8s/addons/logging/fluent-bit/deployed/chart.sls
```

Fixes: #3435

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan

gdemonet commented on salt/metalk8s/addons/logging/fluent-bit/deployed/chart.sls:
```diff
@@ -6,69 +6,32 @@
     heritage: metalk8s
-    release: fluent-bit
-  name: fluent-bit-clusterrole
+  name: fluent-bit
```
Names (for this ClusterRole, and the related binding) change, so we'll have some cleanup to add on upgrade

TeddyAndrieux replied:
Right, I did not check upgrade, I will take a look at it

TeddyAndrieux replied:
Upgrade and downgrade good now ! :smiley:

gdemonet commented on salt/metalk8s/addons/logging/fluent-bit/deployed/chart.sls:
```diff
@@ -6,69 +6,32 @@
 {% raw %}
 
-apiVersion: policy/v1beta1
-kind: PodSecurityPolicy
```
Some objects are removed, so we should remove them on upgrade as well

gdemonet commented on salt/metalk8s/addons/logging/fluent-bit/deployed/configmap.sls:
```diff
@@ -68,40 +68,66 @@ Create fluent-bit ConfigMap:
+                Rename         kubernetes_host node
+                Rename         kubernetes_namespace_name namespace
+                Rename         kubernetes_pod_name pod
+                Remove_wildcard kubernetes_
```
Shouldn't this be: `Remove_wildcard kubernetes_*`?

TeddyAndrieux replied:
No it's not needed

gdemonet commented on salt/metalk8s/addons/logging/fluent-bit/deployed/configmap.sls:
```diff
@@ -68,40 +68,66 @@ Create fluent-bit ConfigMap:
+                Labels         job=fluent-bit
+                Label_Keys     $hostname, $unit
+                Line_Format    json
+                Log_Level      warn
```
Is this doing anything? Like controlling the log level of the plugin? Seems to be only documented for the `[Server]` section :confused:, from what I can see

TeddyAndrieux replied:
It does, you can set the Log_Level per "component"

gdemonet commented on charts/fluent-bit.yaml:
```diff
@@ -11,7 +11,7 @@ podAnnotations:
 resources:
   limits:
-    memory: 100Mi
+    memory: 300Mi
```
:thinking: why are we limiting its memory in the first place? IIRC, `fluentbit` has the tendency to generate memory usage spikes, do we really want it to get OOMKilled if it were to happen? We've seen cases where we had to remove this limit to let the spike pass (was reading too many things on start, not sure why).

TeddyAndrieux replied:
I don't know for the initial reason TBH, I will try to find the reason

TeddyAndrieux replied:
Ok just checked and it's just because it was set by default in the old helm chart value file, but with the new chart it's no longer set by default so I agree to remove it

TeddyAndrieux committed:
salt: Delete the fluent-bit DaemonSet before upgrade and downgrade

Since the label selector for the fluent-bit DaemonSet changed in the
version we embed compare to 2.11 we need to delete this DaemonSet before
upgrading or downgrading it

This commit should be reverted in `development/124.0`

TeddyAndrieux committed:
salt: Cleanup old fluent-bit objects after upgrade and downgrade

Since the fluent-bit helm chart changed, we have some object that get
removed or renamed, so let's clean those right after upgrade and
downgrade

This commit should be reverted in `development/124.0`

gdemonet commented on salt/metalk8s/addons/logging/fluent-bit/deployed/post-downgrade.sls:
```diff
@@ -0,0 +1,31 @@
+        - name: fluent-bit
+        - namespace: metalk8s-logging
+
+{# This one just get renamed from `fluent-bit-clusterrole` to `fluent-bit` #}
+Cleanup new fluent-bit ClusterRole:
```
There was also a `fluent-bit-clusterrolebinding` apparently

TeddyAndrieux replied:
Oh good catch I miss this one :+1:

TeddyAndrieux replied:
Fixed

TeddyAndrieux commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/123.0`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/123.0`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye teddyandrieux.
